Port asp.net core unit testing article to .net core

BillWagner opened the pull request:
The two articles include the overview from the original article, and how to create unit tests in .NET Core using the CLI, and xunit. The portion that covered unit testing in Visual Studio is not a reference to a tutorial on the xunit site.

This depends on PR #430 

Fixes #401

/cc @blackdwarf, @danroth27, @RickAndMSFT, @rachelappel

BillWagner committed:
Add Placeholders for .NET Core Testing articles

BillWagner committed:
Add Placeholders for .NET Core Testing articles

BillWagner committed:
Update TOC for this section, and add placeholders

BillWagner committed:
Update TOC for this section, and add placeholders

BillWagner committed:
caching for a rebase

BillWagner committed:
caching for a rebase

BillWagner committed:
update tests for RC2 folder structure

BillWagner committed:
update tests for RC2 folder structure

BillWagner committed:
ready for a proof read

BillWagner committed:
ready for a proof read

BillWagner committed:
update from proofreading

BillWagner committed:
update from proofreading

svick committed:
Fixed accidental invalid link (#437)

Currently, the line is rendered as:

dnx <a href="for example," title="dnx web">command</a>

which is clearly wrong.

cartermp committed:
Update libraries with cross-platform tools (#412)

This updates the cross-platform library manual.  It removes all of the DNX-isms and changes to guidance to reflect the state of .NET Core for RC2.

cartermp commented on docs/core-concepts/testing/unit-testing-with-dotnet-test.md:
````diff
@@ -0,0 +1,238 @@
+    "xunit": "2.1.0",
+    "dotnet-test-xunit": "1.0.0-rc2-build10015",
+    "PrimeService": "1.0.0"
+```
````
Missing a `}` here.

cartermp commented on docs/core-concepts/testing/unit-testing-with-dotnet-test.md:
````diff
@@ -0,0 +1,238 @@
+You'll also need to add a node to specify the test runner
+at the root of `project.json`:
+
+```json
````
I'd also put enclosing `{` and `}` here.

cartermp commented:
I think you should also call out that you need this to be a `netcoreapp1.0` framework and there are required `imports` for getting xunit to work with the RC2 bits:

```json
"frameworks":{
    "netcoreapp1.0":{
        "imports":[ "dnxcore50", "portable-net45+win8" ]
    }
}
```

That same info is effectively repeated in the libraries doc, but I think it's okay to state it multiple times.

Other than that and the minor feedback I left, this looks great!

Commit:
Fixing documentation on dotnet test (#449)

svick committed:
Missing comma in JSON in dotnet test documentation (#450)

cartermp committed:
Update dotnet pack (#451)

gdoron committed:
Simple space

GiorgioG committed:
Clarified behavior of project dep. resolution (#452)

Spent an extended amount of time trying to debug an issue where we had projects in a subfolder of the 'src' that were not being resolved because dotnet restore does not go beyond one subdirectory of the specified project folder.  In other words, if I have 'src' as my only specified value for 'projects' in global.json, but have projects in 'src/itsOk,', 'src/ItsOk2', 'src/Platforms/ItsNotOk', the ItsNotOk will never be resolved, furthermore the dotnet restore command does not provide any insight into its search behavior (even with debug verbosity.)

cartermp committed:
Merge pull request #453 from gdoron/patch-1

Missing space in RID doc title

BillWagner committed:
updates based on feedback.

BillWagner committed:
updates based on feedback.

BillWagner committed:
Add Placeholders for .NET Core Testing articles (#430)

* Add Placeholders for .NET Core Testing articles

* Update TOC for this section, and add placeholders

BillWagner committed:
hand fixes following a rebase

BillWagner committed:
Merge branch 'UnitTestingInCore' of https://github.com/BillWagner/core-docs into UnitTestingInCore

# Conflicts:
#	docs/core-concepts/testing/unit-testing-with-dotnet-test.md
#	docs/core-concepts/testing/unit-testing.md
